---
title: "Differential gene expression"
output: html_notebook
---

RNAseq data has previously been cleaned and mapped to the reference genome to arrive at counts per gene. This data was saved in a table format.
The data was read into R:

```{r}
counttable = read.table(file = gzfile("data/devspine_read_counts.tsv.gz"), sep = "\t", header = TRUE, row.names = 1)
countmatrix = data.matrix(counttable)
```

PCA was carried out and the result plotted using ggplot2 and ggfortify:
```{r}
require(ggplot2)
require(ggfortify)
pcamatrix <- cbind(condition=c("E12.5", "E12.5", "E12.5", "E16.5", "E16.5", "E16.5", "Adult", "Adult", "Adult"), t(countmatrix))
pca <- prcomp(as.numeric(pcamatrix[, 2:ncol(pcamatrix)]))
autoplot(pca, data = pcamatrix, colour = "condition")
```

