---
jupyter:
  jupytext:
    formats: ipynb,Rmd
    text_representation:
      extension: .Rmd
      format_name: rmarkdown
      format_version: '1.1'
      jupytext_version: 1.2.4
  kernelspec:
    display_name: R
    language: R
    name: ir
---

# Honours Project - Differential gene expression (edgeR)
Schurch et al. have shown that for a low number of replicates, edgeR exact is the preferred tool to use.\
Additionally, a fold change threshold of 1 should be applied. (1)
1. Schurch, N. J., Schofield, P., Gierliński, M., Cole, C., Sherstnev, A., Singh, V., … Barton, G. J. (2016).\
How many biological replicates are needed in an RNA-seq experiment and which differential expression tool should you use? RNA, 22(6), 839. https://doi.org/10.1261/RNA.053959.115

```{r}
countdf <- readRDS("data/devspine_counts_excluded.rds")
TPM <- readRDS("data/TPM.rds")
```

```{r}
highExpr <- function(vector, n1, n2) {
    m1 <- mean(vector[1:n1])
    m2 <- mean(vector[n1+1:n2])
    return(max(c(m1, m2)))
}
```

```{r}
maxTPM <- apply(TPM, 1, highExpr, 3, 3)
```

```{r}
require(edgeR)
require(stringr)
cond <- str_sub(colnames(countdf),1,5)
dge <- DGEList(countdf, group=cond) #create DGEList object containing data from E12.5 and Adult
keep <- filterByExpr(dge) #determine genes with low counts to be excluded
dge <- dge[keep, , keep.lib.sizes=FALSE] #remove genes with low counts
dge <- calcNormFactors(dge, method="TMM") #normalise samples against each other
design <- model.matrix( ~dge$sample$group) #create model (which samples are compared against which)
dge <- estimateDisp(dge, design)
ext <- exactTest(dge) #apply test
all <- topTags(ext, n=nrow(ext$table), adjust.method="BY", sort.by="none")
all <- merge(all, maxTPM, by=0, all.x=TRUE, all.y=FALSE, sort=FALSE)
names(all)[names(all) == 'y'] <- 'maxTPM'
sig <- decideTestsDGE(ext, adjust.method="BY", p.value=0.01, lfc=1)
sig <- all[sig != 0,]
all <- all[order(all$PValue),]
```

```{r}
head(sig[order(-sig$maxTPM),])
```

```{r}
require(ggrepel)
ggplot(data=all, aes(logFC, -log(PValue))) + geom_point(aes(color=log(maxTPM))) +
        scale_colour_viridis_c() + geom_text_repel(data=head(all, 7), aes(label=maxTPM)) +
        coord_cartesian(xlim = c(-20, 20)) + coord_cartesian(ylim = c(-10, 10000))
```
