---
jupyter:
  jupytext:
    formats: ipynb,Rmd
    text_representation:
      extension: .Rmd
      format_name: rmarkdown
      format_version: '1.1'
      jupytext_version: 1.2.4
  kernelspec:
    display_name: R
    language: R
    name: ir
---

# Honours Project - Sanity Checks
RNAseq data has previously been cleaned and mapped to the reference genome to arrive at counts per gene. This data was saved in a table format.
The data was read into R:

```{r}
counttable = read.table(file = gzfile("data/devspine_read_counts.tsv.gz"), sep = "\t", header = TRUE, row.names = 1)
countdf = data.frame(counttable)
```

PCA was carried out and the result plotted using ggplot2 and ggfortify:

```{r}
require(ggplot2)
require(ggfortify)
require(stringr)
Condition=str_sub(colnames(countdf),1,5)
pcadf <- cbind.data.frame(Condition, t(countdf))
pca <- prcomp(pcadf[, 2:ncol(pcadf)])
pcaplot <- autoplot(pca, data = pcadf, colour = "Condition", width=6, height=1)
pcaplot
```

```{r}
ggsave(file="plots/sanity_checks/pca.svg", plot=pcaplot, width=8, height=1.5)
```

The results of this show a clear separation of E12.5 and Adult data. however, the E16.5 replicates do not form a clear cluster.\
For additional analysis, a cluster dendogram was also plotted:

```{r}
require(pvclust)
pvc <- pvclust(countdf)
dendrogram <- plot(pvc)
```

```{r}
svg(filename="plots/sanity_checks/dendrogram.svg", 
    width=5, 
    height=4, 
    pointsize=8)
plot(pvc)
dev.off()
```

In the cluster dendrogram, a separation into two clusters can be observed.\
The first of these clusters includes the three adult samples and one of the E16.5 samples, whereas the second cluster includes the three E12.5 and two of the E16.5 clusters.\
Because of these issues, it was decided to exclude the E16.5 samples for now.\
Additionally, the spike-ins were also removed.

```{r}
saveRDS(countdf[!startsWith(rownames(countdf), "ercc"),-c(4,5,6)], file = "data/devspine_counts_excluded.rds")
```
